### PR TITLE
feat: Add support for ALAC codec

### PR DESCRIPTION
**Pull request by wjywbs**

### CODE REVIEW

**Comment by joeyparrish** [comment]
Please:

1. Merge with `main`, which has changed a lot in the recent `cmake` port
2. Add/update tests

Thanks!

**Comment by cosmin** [comment]
added integration test

File: packager/media/base/stream_info.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ enum Codec {
   kCodecAAC = kCodecAudio,
   kCodecAC3,
   kCodecAC4,
+  kCodecALAC,
```
Comment (joeyparrish) [inline packager/media/base/stream_info.h:44]: I was unsure, so I double-checked.  This is outside the shared library interface, so changing the numbers here won't break users of the library.  It also doesn't impact the format of --output_media_info.

**Review by joeyparrish — APPROVED** [review]

File: packager/media/formats/mp4/box_definitions.cc [hunk 2]
```diff
@@ -2036,6 +2058,7 @@ size_t AudioSampleEntry::ComputeSizeInternal() {
          sizeof(samplesize) + sizeof(samplerate) + sinf.ComputeSize() +
          esds.ComputeSize() + ddts.ComputeSize() + dac3.ComputeSize() +
          dec3.ComputeSize() + dops.ComputeSize() + dfla.ComputeSize() +
+         dac4.ComputeSize() + mhac.ComputeSize() + alac.ComputeSize() +
          dac4.ComputeSize() + mhac.ComputeSize() + udts.ComputeSize() +
```
Comment (wjywbs) [inline packager/media/formats/mp4/box_definitions.cc:2062]: Hi @cosmin, are `dac4` and `mhac` added twice here?

Comment (cosmin) [reply packager/media/formats/mp4/box_definitions.cc:2062]: welp, that's embarrassing. Thanks for catching this.

**Pull request merged**